### PR TITLE
Word embedding2

### PR DESCRIPTION
**Pull request by damianib**
Add word2vec model
Basically a vectorial model, but with a notion of similarity
Details in the code <3

### CODE REVIEW

**Comment by Raklyon** [comment]
Given that you are sing a new package you may want to specify it in the README file

Also, main file runs, which is good, however, I can't tell if the results are correct :innocent:

File: indexing/train_word2vec_model.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        parsed_document = parse_document(absolute_path)
+        corpus.append(parsed_document)
+    print(len(corpus))
+    start = time.time()
+    model = Word2Vec(corpus, min_count=5, size=100, workers=4, window=5, iter=10)
+    print('Training time :', time.time() - start)
+    return model
+
+
+def restrict_w2v(w2v, restricted_word_set):
```
Comment (Hackatosh) [inline indexing/train_word2vec_model.py:24]: Missing types !

File: indexing/train_word2vec_model.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+def clip_google_word2vec_model(collection_directory: str, path_to_google_model: str) -> Word2Vec:
+
+    """Clip the Google word2vec model (far too heavy) so it fits just our collection"""
+
+    file_paths = generate_file_paths(collection_directory)
+    vocabulary = set()
+    for (absolute_path, relative_path) in tqdm.tqdm(file_paths):
+        parsed_document = parse_document(absolute_path)
+        for word in parsed_document:
+            if not(word in vocabulary):
```
Comment (Hackatosh) [inline indexing/train_word2vec_model.py:57]: Why use a set if you perform the check yourself ? What's the point ?

File: indexing/train_word2vec_model.py [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    w2v.index2word = new_index2entity
+
+
+def clip_google_word2vec_model(collection_directory: str, path_to_google_model: str) -> Word2Vec:
+
+    """Clip the Google word2vec model (far too heavy) so it fits just our collection"""
+
+    file_paths = generate_file_paths(collection_directory)
+    vocabulary = set()
+    for (absolute_path, relative_path) in tqdm.tqdm(file_paths):
```
Comment (Hackatosh) [inline indexing/train_word2vec_model.py:54]: What is exactly this tqdm.tqdm doing ? Why is it needed ?

File: indexing/train_word2vec_model.py [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    w2v.vectors = np.array(new_vectors)
+    w2v.index2entity = new_index2entity
+    w2v.index2word = new_index2entity
+
+
+def clip_google_word2vec_model(collection_directory: str, path_to_google_model: str) -> Word2Vec:
+
+    """Clip the Google word2vec model (far too heavy) so it fits just our collection"""
+
+    file_paths = generate_file_paths(collection_directory)
```
Comment (Hackatosh) [inline indexing/train_word2vec_model.py:52]: Maybe we should have an helper "create_vocabulary" or something

File: indexing/train_word2vec_model.py (hunk 3 above)
Comment (damianib) [reply indexing/train_word2vec_model.py:54]: It displays a progress bar
The whole script is taking around 10 minutes to finish so its nice to see the progress

File: indexing/train_word2vec_model.py (hunk 4 above)
Comment (damianib) [reply indexing/train_word2vec_model.py:52]: I only use it in that function so...

File: indexing/train_word2vec_model.py (hunk 2 above)
Comment (damianib) [reply indexing/train_word2vec_model.py:57]: Good point

File: indexing/train_word2vec_model.py (hunk 4 above)
Comment (Hackatosh) [reply indexing/train_word2vec_model.py:52]: Yes but it is not very specific and can be reused, so it should be refactored out of this specific function. I guess :p

**Comment by damianib** [comment]
Pls review again <3

**Pull request merged**